### PR TITLE
Pomodoro timer

### PR DESCRIPTION
**Pull request by bhushannaikaroskar**
Please review my PR for pomodoro timer. Please check the below files.

[Click here](https://github.com/bhushannaikaroskar/pomodoro-app/pull/4/files?file-filters%5B%5D=.js&file-filters%5B%5D=.jsx&show-viewed-files=true)

Timer
pomodoro-timer

### CODE REVIEW

**Review by archit7448 — APPROVED** [review]
clean code , nice UI looks good to merge

File: src/components/edit-task/EditTask.jsx [hunk 1]
```diff
@@ -42,15 +37,16 @@ export default function EditTask({toggleModal,id,title:titleValue,description:de
...
+                description:editData.description,
+                time:editData.time,
+                isCompleted:editData.isCompleted,
             },
         });
-        toggleModal();
-        resetTask();
+        dispatchEditData({type:"RESET"})
+        // toggleModal();
+        // resetTask();
```
Comment (zoef-yt) [inline src/components/edit-task/EditTask.jsx]: remove commented code.

File: src/components/timer/Timer.jsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                        remainingTime={remainingTime}
+                        originalTime={time}
+                    />
+                )}
+            </CountdownCircleTimer>
+            <div className="timer-cta">
+                <button style={!isPlaying?playingStyle:{}} className="timer-button" onClick={pause}>
+                    <span class="material-icons btn-icon-lg">pause</span>
+                </button>
+                <button style={isPlaying?playingStyle:{}} className="timer-button play-button" onClick={play}>
```
Comment (zoef-yt) [inline src/components/timer/Timer.jsx:68]: You can try `isPlaying&&playingStyle`

File: src/components/taskpage/TaskPage.jsx [hunk 3]
```diff
@@ -52,16 +43,14 @@ export default function TaskPage() {
...
                             />
                         );
                     })}
                 </div>
             </div>
             {isModal ? <AddTask toggleModal={toggleModal} /> : ""}
-            {isEditTask ? (
-                <EditTask toggleModal={()=>setIsEditTask(s=>!s)} {...taskObject} />
+            {editData.id ? (
+                <EditTask  />
```
Comment (zoef-yt) [inline src/components/taskpage/TaskPage.jsx]: here you can use "&&" operator as you are not setting anything on the other side of the ternary

File: src/components/edit-task/EditTask.jsx [hunk 4]
```diff
@@ -42,15 +37,16 @@ export default function EditTask({toggleModal,id,title:titleValue,description:de
...
-                id,
-                title,
-                description,
-                time,
-                isCompleted,
+                id:editData.id,
+                title:editData.title,
+                description:editData.description,
+                time:editData.time,
+                isCompleted:editData.isCompleted,
```
Comment (zoef-yt) [inline src/components/edit-task/EditTask.jsx]: destruct edit data.

File: src/components/task-details-page/TaskDetailsPage.jsx [hunk 5]
```diff
@@ -1,21 +1,22 @@
...
 export default function TaskDetailPage() {
     const { taskId } = useParams();
+    console.log(taskId)
     const { taskList, dispatchTaskList } = useTasks();
     const navigate = useNavigate();
 
     const getTaskDetails = (list, id) => {
-        return list.find((task) => task.id === id);
+        return list.find((task) => {
+            return task.id === id});
```
Comment (zoef-yt) [inline src/components/task-details-page/TaskDetailsPage.jsx]: you don't need 2 return the keyword arrow will return the expression.

**Review by zoef-yt — CHANGES_REQUESTED** [review]

File: src/components/timer/Timer.jsx (hunk 2 above)
Comment (bhushannaikaroskar) [reply src/components/timer/Timer.jsx:68]: This won't work as style requires object and this statement return false as boolean

File: src/components/taskpage/TaskPage.jsx (hunk 3 above)
Comment (bhushannaikaroskar) [reply src/components/taskpage/TaskPage.jsx]: done

File: src/components/edit-task/EditTask.jsx (hunk 4 above)
Comment (bhushannaikaroskar) [reply src/components/edit-task/EditTask.jsx]: thanks for this

File: src/components/task-details-page/TaskDetailsPage.jsx (hunk 5 above)
Comment (bhushannaikaroskar) [reply src/components/task-details-page/TaskDetailsPage.jsx]: done

**Review by zoef-yt — APPROVED** [review]
As you have done the suggested changes you are good to merge. Good work Bhushan.

**Pull request merged**